### PR TITLE
Add toxic waste map extra

### PR DESCRIPTION
**Pull request by NetSysFire**
#### Summary
Content "Add toxic waste map extra"

#### Purpose of change

Take 2 of #55211. This adds an illegal toxic waste dump where some random company dumped their trash, which includes chemicals. Rarely however, this company might be XEDRA...

image

#### Describe the solution

The map extra is basically a pool of sewage with normal or trashy rubble ontop with barrels containing mostly vile stuff scattered. There is an okay chance to get a small amount of some chemicals from the barrels. There is also a big variety of chemical spills, which are basically cosmetic only.

Furthermore, there are three "optional" features which the map extra may feature depending on how the RNG feels:

- Radioactive waste (20% chance), to make it fairer and less surprising to the player it will have a broken fence around it and plutonium slurry.
- Toxic gas vents (90% chance to have at least one of them).
- Mutant corpses (10% chance to encounter at least one), lore-stuff: lab staff dumped them there and may also conveniently forget a note with vague clues where to find labs.

#### Describe alternatives you've considered

I am open to changing the chances of some stuff. I am also not 100% sure about the chemicals because I am a noob when it comes to that.

I also thought of adding empty and/or damaged atomic batteries and stuff like atomic lamps to the radioactive part.

#### Testing

Loads correctly, manually explored the world until I found a natural spawn.

#### Additional context

I might subject the non-illegal toxic waste dump to some love, too. So it is only radioactive when plutonium and stuff is being spawned.
Also, I think `liquids` does not work with item groups which is a shame really or copy-pasting the liquids into the toxic gas vent would not be necessary, if setting fields would not erase spilled liquids which is a bug I still have to report.

### CODE REVIEW

File: data/json/items/newspaper.json [hunk 1]
```diff
@@ -53,6 +53,14 @@
     "copy-from": "survnote",
     "snippet_category": "mutant_kid_boss"
   },
+  {
+    "type": "GENERIC",
+    "id": "toxic_waste_labnote",
+    "copy-from": "survnote",
+    "name": { "str": "forgotten note" },
+    "description": "This grimed note seems out of place, as if someone lost it.",
```
Comment (chaohedgehogs) [inline data/json/items/newspaper.json:61]:
```suggestion
    "description": "This grimy note seems out of place, as if someone lost it.",
```

File: data/json/overmap/map_extras.json [hunk 2]
```diff
@@ -547,6 +547,16 @@
     "generator": { "generator_method": "map_extra_function", "generator_id": "mx_reed" },
     "min_max_zlevel": [ 0, 0 ]
   },
+  {
+    "id": "mx_toxic_waste",
+    "type": "map_extra",
+    "name": { "str": "Toxic waste" },
+    "description": "Someone deposed of some toxic waste here.",
```
Comment (chaohedgehogs) [inline data/json/overmap/map_extras.json:554]:
```suggestion
    "description": "Someone dumped some toxic waste here.",
```

File: data/json/snippets/snippets.json [hunk 3]
```diff
@@ -347,5 +347,27 @@
...
       "You're feeling even more paranoid than usual."
     ]
+  },
+  {
+    "type": "snippet",
+    "category": "toxic_waste_corpse_disposal",
+    "text": [
+      {
+        "id": "lost_note_1",
+        "text": "A crumpled and dirty note containing some printed text.  The text contains detailed instructions how to dispose of \"experiments\" in remote areas."
```
Comment (chaohedgehogs) [inline data/json/snippets/snippets.json:357]:
```suggestion
        "text": "A crumpled and dirty note containing some printed text.  The text contains detailed instructions on how to dispose of \"experiments\" in remote areas."
```

File: data/json/snippets/snippets.json [hunk 4]
```diff
@@ -347,5 +347,27 @@
...
+      },
+      {
+        "id": "lost_note_2",
+        "text": "You see a faded map, stained with unknown fluids and riddled with holes.  A route was scribbled onto it.  The point of departure looks to be some kind of odd research facility."
```
Comment (chaohedgehogs) [inline data/json/snippets/snippets.json:361]:
```suggestion
        "text": "You see a faded map, stained with unknown fluids and riddled with holes.  A route is scribbled onto it.  The point of departure looks to be some kind of odd research facility."
```

File: data/json/snippets/snippets.json [hunk 5]
```diff
@@ -347,5 +347,27 @@
...
+      },
+      {
+        "id": "lost_note_3",
+        "text": "Some sheets of paper, folded, crinkled and wet due to exposure to the environment.  It smells vile and there are some faint blood smears.  A map is printed on it, depicting instructions on how to handle deceased \"experiments\" and a route to some remote site.  You find it very odd that the starting point of the route is a subway station."
```
Comment (chaohedgehogs) [inline data/json/snippets/snippets.json:365]:
```suggestion
        "text": "Some sheets of paper, folded, crinkled and wet due to exposure to the environment.  It smells vile and there are faint smears of blood.  A map is printed on it, depicting instructions on how to handle deceased \"experiments\" and a route to some remote site.  You find it very odd that the starting point of the route is a subway station."
```

**Comment by NetSysFire** [comment]
I rebased and yet this damn required test is still failing. Why? I do not see anything related to my PR in the test output, so I have the feeling it may just be one of those random failures again

**Comment by NetSysFire** [comment]
(╯°□°)╯︵ ┻━┻

The reason the test failed was because the survivor note in the same file, which I have not touched, had a "non-sensical item density" and I did a copy-from from it. Super annoying, I had to search for a fix for something I have not touched.

**Comment by NetSysFire** [comment]
I know I am being (literally, heh) pushy, but just a FYI that this is ready to merge unless someone objects due to balance reasons.

**Pull request merged**